functions/Set-DbaSpConfigure.ps1

```diff
@@ -52,78 +55,91 @@ Set-DbaSpConfigure -SqlServer localhost -configs XPCmdShellEnabled -value 1 -Wha
-
-            
-            If ($Pscmdlet.ShouldProcess($SqlServer, "Adjusting server configuration $($configs) from $($currentValue) to $($value)."))
-            {
-                try
-                {
-                    $server.Configuration.$configs.ConfigValue = $value;
-                    $server.Configuration.Alter();
-
-                    #If it's a dynamic setting we're all clear, otherwise let the user know that SQL needs to be restarted for the change to take
-                    if ($isDynamic -eq $true)
-                        {
-                            Write-Output "Config for $($configs) changed. Old value: $($currentRunValue)  New Value: $($value)"
-                        }
-                    else
-                        {
-                            Write-Warning "Config set for $($configs), but restart of SQL Server is required for the new value ($($value)) to be used (old value: $($value))"
-                        }
-
-                }
-                catch
-                {
-                    Write-Error "Unable to change config setting - $($Error)"
-                }
-            }
+			#Grab the current config value
+			$currentValues = ($server.Configuration.$configs)
+			$currentRunValue = $currentValues.RunValue
+			$minValue = $currentValues.Minimum
+			$maxValue = $currentValues.Maximum
+			$isDynamic = $currentValues.IsDynamic
 			
+			#Let us not waste energy setting the value to itself
+			if ($currentRunValue -eq $value)
+			{
+				Stop-Function -Message "Value to set is the same as the existing value. No work being performed."
+			}
+			
+			#Going outside the min/max boundary can be done, but it can break SQL, so I don't think allowing that is wise at this juncture
+			if ($value -le $minValue -or $value -gt $maxValue)
+			{
+				Stop-Function -Message "Value out of range for $configs (min: $minValue - max $maxValue)"
+			}
+			
+			If ($Pscmdlet.ShouldProcess($SqlInstance, "Adjusting server configuration $configs from $currentValue to $value."))
+			{
+				try
+				{
+					$server.Configuration.$configs.ConfigValue = $value
+					$server.Configuration.Alter()
+					
+					[pscustomobject]@{
+						ComputerName = $server.NetName
+						InstanceName = $server.ServiceName
+						SqlInstance = $server.DomainInstanceName
+						OldValue = $currentRunValue
+						NewValue = $value
+					}
+					
+					#If it's a dynamic setting we're all clear, otherwise let the user know that SQL needs to be restarted for the change to take
+					if ($isDynamic -eq $false)
+					{
+						Write-Message -Level Warning -Message "Config set for $configs, but restart of SQL Server is required for the new value ($value) to be used (old value: $($value))"
+					}
+				}
+				catch
+				{
+					Write-Message -Level Warning -Message "Unable to change config setting - $error"
```

Comment:
Use "$_" instead of "$error" - may b a long list otherwise.
Add parameter '-ErrorRecrod $_' so it includes a proper record
Add parameter '-Target $instance' so it becomes easier to troubleshoot